src/MSBuildBinLogQuery/Construction/GraphBuilder.cs

```diff
@@ -34,8 +31,6 @@ public void HandleEvents(params BuildEventArgs[] buildEvents)
 
         private void ProjectStarted(object sender, ProjectStartedEventArgs args)
         {
-            _mutex.WaitOne();
-
             var id = args.BuildEventContext.ProjectInstanceId;
 
             if (!Projects.ContainsKey(id))
```

Comment:
It's not usually recommended to do this kind of check when using a `ConcurrentDictionary`. Instead, use [`GetOrAdd`](https://docs.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2.getoradd?view=netframework-4.8#System_Collections_Concurrent_ConcurrentDictionary_2_GetOrAdd__0_System_Func__0__1__) to atomically add the new project if an equivalent one doesn't exist.

Reply:
Done.